src/sinks/util/tcp.rs

```diff
@@ -225,13 +241,20 @@ impl Sink for TcpSink {
                         emit!(TcpEventSent {
                             byte_size: line.len()
                         });
+                        let line = Bytes::copy_from_slice(&line);
```

Comment:
This a conversion between two `bytes` crates in unfortunate. The extra copies are probably not that big of a hit to the performance, so I guess it's ok. Just pointing it out.

Reply:
Yeah. The bad part of such slow update that in the future we will need to find and rid all such conversions. Also new `bytes` have extra copying on converting between Bytes <=> BytesMut :disappointed:

Reply:
True, yet it seems kind of odd that `Bytes` were convertible to `BytesMut` before in the first place. The API seemes to have been incorrect, thus we're probably using it incorrectly... We probably should do an audit of the code as part of old `bytes` removal - I bet there are a lot of places where we can avoid copying where we could just switch to `Bytes` to `BytesMut`.

Reply:
An interesting thing, will `clippy` catch extra operations with Bytes or not. I already started improving `bytes` from BytesMut => Bytes through freeze, but this pretty hard for my current Rust level. https://github.com/tokio-rs/bytes/pull/418